lib/pages/chat_draft/draft_chat.dart

```diff
@@ -356,26 +346,62 @@ class DraftChatController extends State<DraftChat>
         )
         .toList();
 
-    await sendImagesWithCaption(
+    _handleSendFileOnWeb(context, matrixFilesList);
+  }
+
+  Future<void> _handleSendFileOnWeb(
+    BuildContext context,
+    List<MatrixFile> matrixFilesList,
+  ) async {
+    const int maxFileQuantity = 1;
+    if (matrixFilesList.length > maxFileQuantity) {
+      TwakeSnackBar.show(
+        context,
+        L10n.of(context)!.countFilesSendPerDialog(maxFileQuantity),
+      );
+      return;
+    }
+
+    if (matrixFilesList.isEmpty) {
+      TwakeSnackBar.show(
+        context,
+        L10n.of(context)!.failedToSendFiles,
+      );
+      return;
+    }
+
+    final sendFileOnWebInteractor = getIt.get<SendFileOnWebInteractor>();
+
+    final dialogStatus = await sendImagesWithCaption(
       context: context,
       matrixFiles: [matrixFilesList.first],
     );
-    isSendingNotifier.value = true;
-    _createRoom(
-      onRoomCreatedSuccess: (newRoom) {
-        if (matrixFilesList.first is MatrixImageFile) {
-          sendMediaWithCaptionInteractor.execute(
-            room: newRoom,
-            media: matrixFilesList.first,
-          );
-        } else {
-          sendFileOnWebInteractor.execute(
-            room: newRoom,
-            files: matrixFilesList,
+
+    if (dialogStatus is SendMediaWithCaptionStatus) {
+      switch (dialogStatus) {
+        case SendMediaWithCaptionStatus.cancel:
+          break;
+        case SendMediaWithCaptionStatus.done:
+        case SendMediaWithCaptionStatus.error:
```

Comment:
in the case of `SendMediaWithCaptionStatus.error`  why would it behave the same as the case of `SendMediaWithCaptionStatus.done` ?

Reply:
This is because the `error` case stands for `room` is null

Reply:
i think in the case of error, no need to create room, instead of log it only

Reply:
After selected images, the `SendFileDialog` appear. If click on `Send`, it check if the current room is null then pop back with value `error`. In this problem, user is inside a draft chat so the `room` will be null. Therefore, `SendFileDialog` always return `error`. So case `error` also need to create a room or file will never be sent.

Reply:
okay, you're right

Reply:
so i think the best we should do is change `SendMediaWithCaptionStatus.error` -> `SendMediaWithCaptionStatus.emptyRoom`. Because its not error in the case of draft chat, but will be error in case of send Files in chat

Reply:
IMO: Separate func to handler result of dialog.
- Easy to understanding 
- Easy to handle result

Reply:
Updated!